src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/flowunit/HighHeapUsageClusterFlowUnit.java

```diff
@@ -0,0 +1,32 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.flowunit;
```

Comment:
I still don't quite understand why do we need to create a derived flowunit class for each RCA type. I mean, even if we have different types of RCAs, the ResourceFlowUnit itself should be the same and we are not expected to add additional variables/method to the flowunit class ?

Reply:
There are two reasons:
1. By specifying a type for a node, we gain more safety against wrongly assigning flowunits across different nodes.
2. The ResourceFlowUnit is pretty basic and can only do so much. If you need to pass more information in your resource flow unit there is no way right now. So, with this change, we can let users defining the graph also define what they want interchanged between nodes for each vertex.

Also, if someone does not want to define a new derived flow unit type, they can still continue to use ResourceFlowUnit as the type and it will still work with the limitations imposed by that particular flow unit.

The necessity to specify flow unit type per vertex will become clearer with the next PR which makes each flowunit type a protobuf message to facilitate a single wrapper NodeFlowUnitMessage which can contain any flowunit inner message.

Reply:
Ok, now I can understand this a bit better. But if each node has a derived type of flowunit, will they be persisited into SQL differently (using different schema)?
The reason why I want to ask is I just want to make sure we are not working on the same thing and changing the flowunit simulatously as we are currently working on two diverged branchs. The way I did in my code to persist the flowunit if to create abstract metheds : 

public List<Field<?>> getSqlSchema()
public List<Object> getSqlValue()

and implement those methods in each flowunit

Reply:
Yes, they will have to be persisted differently. This model works well with the strategy you already have for persisting flow units. I will be able to reuse the methods you have defined to persist.

We can discuss merge strategies offline :)